### PR TITLE
Initialize server side implementation

### PR DESCRIPTION
**Pull request by HitRam**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		else {
+			socket.emit('user exists', name);
+		}
+	});
+
+});
+
+http.listen('3000', function() {
+	console.log('listening on *3000');
+});
```
Comment (abhigyank) [inline app.js]: Leave a newline at the end of file.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+var users = [];
+
+io.on('connection', function(socket) {
+
+	socket.on('set username', function(name) {
+		//username not taken
+		if(users.indexOf(name) == -1) {
+			socket.emit('user set', name);
+		}
```
Comment (abhigyank) [inline app.js:20]: Write code to update users array.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+io.on('connection', function(socket) {
+
+	socket.on('set username', function(name) {
+		//username not taken
+		if(users.indexOf(name) == -1) {
+			socket.emit('user set', name);
+		}	
+		//username taken
+		else {
+			socket.emit('user exists', name);
```
Comment (abhigyank) [inline app.js:23]: Add code to remove this clients used  username from users array.

**Review by abhigyank — CHANGES_REQUESTED** [review]

File: app.js (hunk 2 above)
Comment (HitRam) [reply app.js:20]: Oops sorry! 
Will do it soon.

File: app.js (hunk 3 above)
Comment (HitRam) [reply app.js:23]: Why do you want to remove it?
It would be someone else's username so we want it in users array!

File: app.js (hunk 1 above)
Comment (HitRam) [reply app.js]: Buy me a gitmate 😛

**Review by abhigyank — APPROVED** [review]

**Pull request merged**